kustomize/commands/edit/set/configmap.go

```diff
@@ -0,0 +1,151 @@
+		},
+	}
+
+	cmd.Flags().StringArrayVar(
+		&flags.LiteralSources,
+		util.FromLiteralFlag,
+		[]string{},
+		"Specify an existing key and a new value to update a ConfigMap (i.e. mykey=newvalue)")
+	cmd.Flags().StringVar(
+		&flags.Namespace,
+		util.NamespaceFlag,
+		"",
+		"Current namespace of the target ConfigMap")
+	cmd.Flags().StringVar(
+		&flags.NewNamespace,
+		util.NewNamespaceFlag,
+		"",
+		"New namespace value for the target ConfigMap")
+
+	return cmd
+}
+
+func runEditSetConfigMap(
+	flags util.ConfigMapSecretFlagsAndArgs,
+	fSys filesys.FileSystem,
+	args []string,
+	ldr ifc.KvLoader,
+	rf *resource.Factory,
+) error {
+	err := flags.ExpandFileSource(fSys)
+	if err != nil {
+		return fmt.Errorf("failed to expand file source: %w", err)
+	}
+
+	err = flags.ValidateSet(args)
+	if err != nil {
+		return fmt.Errorf("failed to validate flags: %w", err)
+	}
+
+	// Load the kustomization file.
+	mf, err := kustfile.NewKustomizationFile(fSys)
+	if err != nil {
+		return fmt.Errorf("failed to load kustomization file: %w", err)
+	}
+
+	kustomization, err := mf.Read()
+	if err != nil {
+		return fmt.Errorf("failed to read kustomization file: %w", err)
+	}
+
+	// Updates the existing ConfigMap
+	err = setConfigMap(ldr, kustomization, flags, rf)
+	if err != nil {
+		return fmt.Errorf("failed to create configmap: %w", err)
+	}
+
+	// Write out the kustomization file with added configmap.
+	err = mf.Write(kustomization)
+	if err != nil {
+		return fmt.Errorf("failed to write kustomization file: %w", err)
+	}
+
+	return nil
+}
+
+func setConfigMap(
+	ldr ifc.KvLoader,
+	k *types.Kustomization,
+	flags util.ConfigMapSecretFlagsAndArgs,
+	rf *resource.Factory,
+) error {
+	args, err := findConfigMapArgs(k, flags.Name, flags.Namespace)
+	if err != nil {
+		return fmt.Errorf("could not set new ConfigMap value: %w", err)
+	}
+
+	if len(flags.LiteralSources) > 0 {
+		err := util.UpdateLiteralSources(&args.GeneratorArgs, flags)
+		if err != nil {
+			return fmt.Errorf("failed to update literal sources: %w", err)
+		}
+	}
+
+	// update namespace to new one
+	if flags.NewNamespace != "" {
+		args.Namespace = flags.NewNamespace
+	}
+
+	// Validate by trying to create corev1.configmap.
+	args.Options = types.MergeGlobalOptionsIntoLocal(
+		args.Options, k.GeneratorOptions)
+
+	_, err = rf.MakeConfigMap(ldr, args)
+	if err != nil {
+		return fmt.Errorf("failed to validate ConfigMap structure: %w", err)
+	}
+
+	return nil
+}
+
+// findConfigMapArgs finds the generator arguments corresponding to the specified
+// ConfigMap name. ConfigMap must exist for this command to be successful.
+func findConfigMapArgs(m *types.Kustomization, name, namespace string) (*types.ConfigMapArgs, error) {
+	cmIndex := slices.IndexFunc(m.ConfigMapGenerator, func(cmArgs types.ConfigMapArgs) bool {
+		return name == cmArgs.Name && namespace == cmArgs.Namespace
+	})
+
+	if cmIndex == -1 {
+		return nil, fmt.Errorf("unable to find ConfigMap with name '%s'", name)
```

Comment:
Just curious, can cmArgs ever be nil? Asking this because `types.Kustomization` is being passed as a pointer. Would it be safer to check if it is nil first and return it before itself?

Reply:
@varshaprasad96 I'd say it's unlikely, because `ConfigMapGenerator` itself is not a pointer (it's a slice within the kustomization file). I'd expect the read for `kustomization.yaml` to fail before we can have a `nil` kustomization object, but to be honest I can't picture a situation where the evaluated value of the object in `kustomization.yaml` would result in a `nil` slice off the top of my head.

Do you have a specific use case and/or failure mode in mind? I'm happy to add some tests to safeguard against this.

Reply:
Got it! Thanks @stormqueen1990! I was just curious because of the func was using a pointer, but makes sense that as we already have a fail safe knob before :)

Reply:
No problem @varshaprasad96! Do you feel this is resolved? Gauging to see if I can click "resolve conversation" 😃

Reply:
Yes, resolved from my end :)